src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs

```diff
@@ -51,11 +52,33 @@ internal static void GetDefaultIdentityAndHash(out object identity, out string h
             Assembly? assembly = Assembly.GetEntryAssembly();
 
             if (assembly == null)
+            {
                 throw new IsolatedStorageException(SR.IsolatedStorage_Init);
+            }
+
+            // in case of SingleFile, Location.Length is 0, we will use native entrypoint's filename.
+            if (assembly.Location.Length == 0)
+            {
+                using (Process currentProcess = Process.GetCurrentProcess())
```

Comment:
These API is not supported in browser, so this won't fix the wasm case. (It may be even breaking it.)

Reply:
This flow is safe because:
1. IsolatedSotrage is currently not supported on wasm.
2. Assembly.GetEntryAssembly() returns null, so it will not reach here, instead it will throw `IsolatedStorageException` from (line 56).